src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -1002,8 +1120,7 @@ public long getTrueAllocatedSizeIn(File input)
 
     public static List<File> getKSChildDirectories(String ksName)
     {
-        return getKSChildDirectories(ksName, dataDirectories);
-
+        return getKSChildDirectories(ksName, dataDirectories.getDataDirectoriesUsedBy(ksName));
```

Comment:
Maybe the two versions of `getKSChildDirectories` can be merged, given that one is the only caller of the other?